crm_claim_rma/crm_claim_rma.py

```diff
@@ -0,0 +1,584 @@
+                invoice, claim_type, product, claim_date)
+        except (InvoiceNoDate, ProductNoSupplier):
+            # we don't mind at this point if the warranty can't be
+            # computed and we don't want to block the user
+            values.update({'guarantee_limit': False, 'warning': False})
+        else:
+            values.update(warranty)
+
+        warranty_address = claim_line_obj._warranty_return_address_values(
+            product, company, warehouse)
+        values.update(warranty_address)
+
+        self.update(values)
+
+    def _warranty_return_address_values(self, product, company, warehouse):
+        """Return the partner to be used as return destination and
+        the destination stock location of the line in case of return.
+
+        We can have various case here:
+            - company or other: return to company partner or
+              crm_return_address_id if specified
+            - supplier: return to the supplier address
+
+        """
+        if not (product and company and warehouse):
+            return {'warranty_return_partner': False,
+                    'warranty_type': False,
+                    'location_dest_id': False}
+        sellers = product.seller_ids
+        if sellers:
+            seller = sellers[0]
+            return_address_id = seller.warranty_return_address.id
+            return_type = seller.warranty_return_partner
+        else:
+            # when no supplier is configured, returns to the company
+            return_address = (company.crm_return_address_id or
+                              company.partner_id)
+            return_address_id = return_address.id
+            return_type = 'company'
+        location_dest_id = self.get_destination_location(product, warehouse)
+        return {'warranty_return_partner': return_address_id,
+                'warranty_type': return_type,
+                'location_dest_id': location_dest_id}
+
+    def set_warranty_return_address(self):
+        claim = self.claim_id
+        product = self.product_id
+        company = claim.company_id
+        warehouse = claim.warehouse_id
+        values = self._warranty_return_address_values(
+            product, company, warehouse)
+        self.write(values)
+        return True
+
+    @api.multi
+    def set_warranty(self):
+        """ Calculate warranty limit and address """
+        for claim_line in self:
+            if not (claim_line.product_id and claim_line.invoice_line_id):
+                raise Warning(
+                    _('Error'), _('Please set product and invoice.'))
+            claim_line.set_warranty_limit()
+            claim_line.set_warranty_return_address()
+
+
+# TODO add the option to split the claim_line in order to manage the same
+# product separately
+class CrmClaim(Model):
+    _inherit = 'crm.claim'
+
+    def init(self, cr):
+        cr.execute("""
+            UPDATE "crm_claim" SET "number"=id::varchar
```

Comment:
It's the way to make one task in module installation. It's done in the module I mention that I recommend you to use instead of rewriting this: https://github.com/OCA/crm/blob/8.0/crm_claim_code/__init__.py#L10 and declare here: https://github.com/OCA/crm/blob/8.0/crm_claim_code/__openerp__.py#L48